src/applications/bmqstoragetool/m_bmqstoragetool_parameters.h

```diff
@@ -91,19 +104,37 @@ struct CommandLineArguments {
 
     // MANIPULATORS
     /// Validate the consistency of all settings.
-    bool validate(bsl::string* error);
+    bool validate(bsl::string* error, bslma::Allocator* allocator = 0);
 };
 
 struct Parameters {
+    // PUBLIC TYPES
+    enum SearchValueType {
+        e_NONE         = 0,
+        e_TIMESTAMP    = 1,
+        e_SEQUENCE_NUM = 2,
+        e_OFFSET       = 3
+    };
+
     // PUBLIC DATA
     QueueMap d_queueMap;
     // Queue map containing uri to key and key to info mappings
-    bsls::Types::Int64 d_timestampGt;
-    // Filter messages by minimum timestamp
-    bsls::Types::Int64 d_timestampLt;
-    // Filter messages by maximum timestamp
+    SearchValueType d_valueType;
+    // Search value type for message filtering
+    bsls::Types::Uint64 d_valueGt;
+    // Filter messages greater than value
+    bsls::Types::Uint64 d_valueLt;
+    // Filter messages less than value
+    CompositeSequenceNumber d_seqNumGt;
+    // Filter messages greater than sequence number
+    CompositeSequenceNumber d_seqNumLt;
+    // Filter messages less than sequence number
```

Comment:
This change looks confusing: There are 3 pairs of parameters, we store two of them in this struct and use only one pair. So let's keep all the three. We may need them in the future if our customers ask us to enable combined filtering. Moreover it will be really easy to implement.

Reply:
Agree, moved range parameters to `Parameters::Range` struct.